### PR TITLE
Config Generator should not return raw errors via the REST API

### PR DESCRIPTION
**Pull request by smarterclayton**
Returning non api.Status errors is verboten

@ironcladlou review plx

### CODE REVIEW

File: pkg/deploy/generator/config_generator.go [hunk 1]
```diff
@@ -60,12 +59,12 @@ func (g *DeploymentConfigGenerator) Generate(ctx kapi.Context, name string) (*de
...
 	refs, legacy := findReferences(dc)
 	if errs := retrieveReferences(g.Client, ctx, refs, legacy); len(errs) > 0 {
-		return nil, kerrors.NewAggregate(errs)
+		return nil, errors.NewInvalid("DeploymentConfig", name, errs)
 	}
 	indexed := referencesByIndex(refs, legacy)
 	changed, errs := replaceReferences(dc, indexed)
 	if len(errs) > 0 {
-		return nil, kerrors.NewAggregate(errs)
+		return nil, errors.NewInvalid("DeploymentConfig", name, errs)
```
Comment (ironcladlou) [inline pkg/deploy/generator/config_generator.go]: Should this be `deploymentConfig`? See also: `imageRepository`

Comment (smarterclayton) [inline pkg/deploy/generator/config_generator.go]: Your deployment config is invalid (it's pointing to things that don't exist).

Comment (ironcladlou) [inline pkg/deploy/generator/config_generator.go]: I meant just the actual naming/capitalization for a field/type reference.

Comment (smarterclayton) [inline pkg/deploy/generator/config_generator.go]: Kinds should be capitalized, the other spot is wrong and I'll fix it.  I'm only going to fix this file though.

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**